force-app/main/default/lwc/attendance/attendance.js

```diff
@@ -0,0 +1,133 @@
+
+export default class Attendance extends LightningElement {
+    @api recordId;
+    @track serviceDeliveries;
+    @track fieldSet;
+
+    unitOfMeasurement;
+
+    labels = {
+        submit: Submit_Label,
+        attendance: Attendance_Label,
+    };
+
+    fields = {
+        contact: CONTACT_FIELD,
+        quantity: QUANTITY_FIELD,
+        attendanceStatus: ATTENDANCE_STATUS_FIELD,
+        createdDate: CREATED_DATE_FIELD,
+        createdBy: CREATED_BY_FIELD,
+    };
+
+    @wire(getRecord, {
+        recordId: "$recordId",
+        fields: [UNITMEASUREMENT_RELATED_FIELD],
+    })
+    wiredSession(result) {
+        if (!(result.data || result.error)) {
+            return;
+        }
+
+        if (result.data) {
+            let schedule = getChildObjectByName(result.data.fields, "ServiceSchedule__r");
+            let service = getChildObjectByName(schedule.value.fields, "Service__r");
+            this.unitOfMeasurement =
+                service.value.fields[UNITMEASUREMENT_SERVICE_FIELD.fieldApiName].value;
+        } else if (result.error) {
+            handleError(result.error);
+        }
+    }
+
+    @wire(generateRoster, { sessionId: "$recordId" })
+    wiredServiceDeliveries(result) {
+        if (!(result.data || result.error)) {
+            return;
+        }
+
+        if (result.data) {
+            this.serviceDeliveries = [...result.data];
+        } else if (result.error) {
+            handleError(result.error);
+        }
+    }
+
+    @wire(getFieldSet, {
+        objectName: SERVICEDELIVERY_OBJECT.objectApiName,
+        fieldSetName: FIELD_SET_NAME,
+    })
+    wiredFields(result) {
+        if (result.data) {
+            this.configureFieldSet(result.data.map(a => ({ ...a })));
+        } else if (result.error) {
+            handleError(result.error);
+        }
+    }
+
+    get header() {
+        return (
+            this.labels.attendance +
+            (this.serviceDeliveries && this.serviceDeliveries.length
+                ? " (" + this.serviceDeliveries.length + ")"
+                : "")
+        );
+    }
+
+    configureFieldSet(incomingFieldSet) {
+        let finalFieldSet = [];
+        incomingFieldSet.forEach(field => {
+            field.disabled = false;
+            field.isQuantityField = false;
+            field.isOutputField = false;
+            field.isNormalInputField = false;
+            field.isContactField = false;
+
+            // Number fields are size 1
+            // Client lookup is size 3
+            // Everything else is size 2
+            if (SHORT_DATA_TYPES.includes(field.type)) {
+                field.size = 1;
+            } else if (LONG_DATA_TYPES.includes(field.type)) {
+                field.size = 3;
+            } else {
+                field.size = 2;
+            }
+
+            if (field.apiName === this.fields.contact.fieldApiName) {
+                field.isContactField = true;
+                field.size = 2;
```

Comment:
@bethbrains , Isn't this the lookup field which should have a size of 3 and is set on line 111. Should we be setting the size again here.